config-service-impl/src/main/java/org/hypertrace/config/service/store/ConfigDocument.java

```diff
@@ -77,7 +81,8 @@ public ConfigDocument(
       @JsonProperty(TENANT_ID_FIELD_NAME) String tenantId,
       @JsonProperty(CONTEXT_FIELD_NAME) String context,
       @JsonProperty(VERSION_FIELD_NAME) long configVersion,
-      @JsonProperty(USER_ID_FIELD_NAME) String userId,
+      @JsonProperty(LAST_UPDATED_USER_ID_FIELD_NAME) String lastUpdatedUserId,
```

Comment:
so I certainly prefer this name, but given this is an existing field it makes me a little uncomfortable. Two options:

- if there's no data in there and you test this against some existing entries that it gracefully handles defaulting it/ignoring the old empty string, we can go with this. I _think_ it should but wouldn't want to merge without actually trying. And for that matter, if that causes an issue the new one might anyway.
- otherwise, we can rename the code but leave the actual string value that represents the json field name unchanged.

Reply:
Looks like there's both a jackson based and proto based serialization config in here. The proto one should be fine. The jackson one is set to ignore unknown on deserialize so should be fine for the rename
```
new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
```
Only question is how does it default the new ones? May have to set a default annotation.

Reply:
Also, adding a compatibility test would be good. This is unit testable, doesn't need to be manual.